### PR TITLE
Add multiple-column PK tests

### PR DESCRIPTION
**Pull request by birdstorm**
### What problem does this PR solve? 
Add multiple-column PK tests

- Add Date/Timestamp random value generation
- Add 6w+ tests concerning primary keys with multiple columns 

### Check List 

Tests 

 - Integration test

### CODE REVIEW

File: .ci/integration_test.groovy [hunk 1]
```diff
@@ -90,12 +90,13 @@ def call(ghprbActualCommit, ghprbCommentBody, ghprbPullId, ghprbPullTitle, ghprb
                         sh """
                         cp -R /home/jenkins/git/tispark/. ./
                         git checkout -f ${ghprbActualCommit}
-                        find core/src -name '*Suite*' > test
+                        find core/src -name '*Suite*' | grep -v 'MultiColumnPKDataTypeSuite' > test
+                        shuf test -o  test2
+                        mv test2 test
+                        find core/src -name '*MultiColumnPKDataTypeSuite*' >> test
```
Comment (marsishandsome) [inline .ci/integration_test.groovy]: why add `MultiColumnPKDataTypeSuite` to the end?

Comment (birdstorm) [reply .ci/integration_test.groovy]: We use round robin to split all tests evenly (especially `MultiColumnPKDataTypeSuite`, ordering them together ensures that 36 suites are separated evenly in 18 test containers), also they run longer and should be executed later to minimize test waiting period.

File: core/src/test/scala/org/apache/spark/sql/test/generator/TestDataGenerator.scala [hunk 2]
```diff
@@ -279,8 +283,11 @@ object TestDataGenerator {
   }
 
   def hash(value: Any): String = value match {
-    case null           => "null"
-    case b: Array[Byte] => b.mkString("[", ",", "]")
+    case null                  => "null"
+    case b: Array[Byte]        => b.mkString("[", ",", "]")
+    case t: java.sql.Timestamp =>
+      // timestamp was indexed as Integer when treated as unique key
```
Comment (zhexuany) [inline core/src/test/scala/org/apache/spark/sql/test/generator/TestDataGenerator.scala]: on what situations?

File: core/src/test/scala/org/apache/spark/sql/BaseTiSparkTest.scala [hunk 3]
```diff
@@ -166,12 +164,6 @@ class BaseTiSparkTest extends QueryTest with SharedSQLContext {
     loadTestData()
   }
 
-  protected def initializeTimeZone(): Unit = {
```
Comment (zhexuany) [inline core/src/test/scala/org/apache/spark/sql/BaseTiSparkTest.scala]: why do we remove initializeTimeZone ?

Comment (zhexuany) [reply core/src/test/scala/org/apache/spark/sql/BaseTiSparkTest.scala]: Never mind, I just thought you move to another file.

Comment (birdstorm) [reply core/src/test/scala/org/apache/spark/sql/BaseTiSparkTest.scala]: added in SharedSQLContext

File: core/src/test/scala/org/apache/spark/sql/test/generator/TestDataGenerator.scala (hunk 2 above)
Comment (birdstorm) [reply core/src/test/scala/org/apache/spark/sql/test/generator/TestDataGenerator.scala]: unique key, as commented

**Pull request merged**